src/pages/Facility/settings/organizations/components/FacilityOrganizationSelector.tsx

```diff
@@ -39,6 +47,189 @@ interface FacilityOrganizationSelectorProps {
   singleSelection?: boolean;
 }
 
+interface OrganizationPopoverProps {
+  className?: string;
+  t: (key: string) => string;
```

Comment:
why pass t in?

Also, you should be able to move some of these inside the popover (like OrganizationQueries and getCurrentLevelOrganizations, just need to pass in navigationLevels in as props).

When you make a change like this, definitely spend time on thinking on how it could be improved.